src/sortStudents.ts

```diff
@@ -1,16 +1,66 @@
 
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name,
+  Surname,
+  Age,
+  Married,
+  AverageGrade,
 }
 
 // create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const result = [...students];
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  function getAverageGrade(grades: number[]):number {
+    return grades.reduce((total: number, i: number) => total + i, 0)
+      / grades.length;
+  }
+
+  switch (sortBy) {
+    case SortType.Age:
+      return (order === 'asc')
+        ? result.sort((a, b) => a.age - b.age)
+        : result.sort((a, b) => b.age - a.age);
+
+    case SortType.Name:
+      return (order === 'asc')
+        ? result.sort((a, b) => a.name.localeCompare(b.name))
+        : result.sort((a, b) => b.name.localeCompare(a.name));
+
+    case SortType.Surname:
+      return (order === 'asc')
+        ? result.sort((a, b) => a.surname.localeCompare(b.surname))
+        : result.sort((a, b) => b.surname.localeCompare(a.surname));
```

Comment:
we can combine some cases
```suggestion
    case SortType.Name:
    case SortType.Surname:
      return (order === 'asc')
        ? result.sort((a, b) => a[sortBy].localeCompare(b[sortBy]))
        : result.sort((a, b) => b[sortBy].localeCompare(a[sortBy]));
```

(don't forget to update the enum)